app/src/main/java/org/fossasia/openevent/general/about/AboutEventActivity.kt

```diff
@@ -58,11 +58,26 @@ class AboutEventActivity : AppCompatActivity(), AppBarLayout.OnOffsetChangedList
         val dateString = StringBuilder()
         val startsAt = EventUtils.getLocalizedDateTime(event.startsAt)
         val endsAt = EventUtils.getLocalizedDateTime(event.endsAt)
-        aboutEventDetails.text = dateString.append(EventUtils.getFormattedDate(startsAt))
-                                           .append(" - ")
-                                           .append(EventUtils.getFormattedDate(endsAt))
-                                           .append(" • ")
-                                           .append(EventUtils.getFormattedTime(startsAt))
+        if (EventUtils.getFormattedDate(startsAt) != EventUtils.getFormattedDate(endsAt)) {
+            aboutEventDetails.text = dateString.append(EventUtils.getFormattedDate(startsAt))
+                    .append(" at ")
+                    .append(EventUtils.getFormattedTime(startsAt))
+                    .append(" - ")
+                    .append(EventUtils.getFormattedDate(endsAt))
+                    .append(" at ")
+                    .append(EventUtils.getFormattedTime(endsAt))
+                    .append(" ")
+                    .append(EventUtils.getFormattedTimeZone(endsAt))
+        } else {
+            aboutEventDetails.text = dateString.append(EventUtils.getFormattedDate(startsAt))
+                    .append(" from ")
+                    .append(EventUtils.getFormattedTime(startsAt))
+                    .append(" to ")
+                    .append(EventUtils.getFormattedTime(endsAt))
+                    .append(" (")
+                    .append(EventUtils.getFormattedTimeZone(endsAt))
+                    .append(")")
```

Comment:
May be you can encapsulate this somewhere in a helper function. You may need it else where. Even if you don't need it then also there is no need to pollute the Activity with such logic.